spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -319,11 +320,20 @@ else if (pageable instanceof DatastorePageable) {
 	private void applySelectWithFilter(Object[] parameters, Builder builder) {
 		Iterator it = Arrays.asList(parameters).iterator();
 		Filter[] filters = this.filterParts.stream().map((part) -> {
-			DatastorePersistentProperty persistentProperty = (DatastorePersistentProperty) this.datastorePersistentEntity
-					.getPersistentProperty(part.getProperty().getSegment());
+			Iterable<PropertyPath> iterable = () -> part.getProperty().iterator();
+			List<DatastorePersistentProperty> properties =
+					StreamSupport.stream(iterable.spliterator(), false)
+					.map(propertyPath -> {
+						DatastorePersistentEntity<?> persistentEntity =
+								this.datastoreMappingContext.getPersistentEntity(propertyPath.getOwningType());
+						return persistentEntity.getPersistentProperty(propertyPath.getSegment());
+					}).collect(Collectors.toList());
+
+			String fieldName = properties.stream().map(DatastorePersistentProperty::getFieldName)
```

Comment:
In Firestore you call the method `getPropertyName` but here it's `getFieldName`. Is it too late to make them consistent?